nms/app/views/events/EventsTable.js

```diff
@@ -261,6 +261,8 @@ export default function EventsTable(props: EventTableProps) {
   }, [startDate, endDate]);
 
   useEffect(() => {
+    setStartDate(props.inStartDate); 
```

Comment:
Apart from the formating issues, there are several problems here:

- `props.inStartDate` and  `props.inEndDate` can be undefined. Setting the start and end date to undefined breaks the `EventsTable` this happening if you open the gateway details and crashes the page.
- `setStartDate` and `setEndDate` are missing in the dependencies of useEffect. See the linter warning below. But to be able to add them we need to wrap the callback for `useRefreshingDateRange` into a `useCallback`.
- if start and end date are changed `useRefreshingDateRange` will already refresh the table to there is no need to call `onQueryChange` again.

Reply:
Hi @thmsschmitt, very kind you feedback. I'll check all these points here and I'll update the PR.